src/coprocessor/codec/mysql/time/tz.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str::FromStr;
+
+use chrono::*;
+use chrono_tz;
+
+/// A time zone represented by either offset (i.e. +8) or name (i.e. Asia/Shanghai). In addition,
+/// local time zone is also valid.
+#[derive(Clone, Copy)]
+pub enum Tz {
+    /// A time zone specified by offset seconds.
+    Offset(FixedOffset),
+
+    /// A time zone specified by name (in IANA database).
+    Name(chrono_tz::Tz),
+
+    /// A local time zone.
+    Local(Local),
+    // chrono also has a offset kind `chrono::offset::Utc`, but we only use UTC in tests,
+    // so we don't introduce an extra enum item for it.
+}
+
+impl Tz {
+    /// Constructs a time zone from the offset in seconds.
+    pub fn from_offset(secs: i64) -> Option<Self> {
+        FixedOffset::east_opt(secs as i32).map(Tz::Offset)
+    }
+
+    /// Constructs a time zone from the name. If the specified time zone name is `system`,
+    /// a local time zone will be constructed.
+    pub fn from_tz_name(name: &str) -> Option<Self> {
+        if name.to_lowercase() == "system" {
+            Some(Tz::local())
+        } else {
+            chrono_tz::Tz::from_str(name).ok().map(Tz::Name)
+        }
+    }
+
+    /// Constructs a UTC time zone.
+    pub fn utc() -> Self {
+        Tz::Name(chrono_tz::UTC)
+    }
+
+    /// Constructs a local time zone.
+    pub fn local() -> Self {
+        Tz::Local(Local)
+    }
+}
+
+impl fmt::Debug for Tz {
```

Comment:
I see `Local`, the `Tz`, and  `Offset` all implement `Debug`, why did you not just `#[derive]` it?

Reply:
I don't want to print out something like `Tz::Offset(Asia/Shanghai)`, only want `Asia/Shanghai`.

Reply:
I actually thought about this some more and I think `Tz::Name(Asia/Shanghai)` is a good output for `Debug`, and I think `Display` is more appropriate to produce `Asia/Shanghai`.

What do you think?